app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt

```diff
@@ -61,34 +68,80 @@ class AddPictureActivity : AppCompatActivity() {
         }
     }
 
-    private fun cameraIsAvailable(): Boolean {
-        return applicationContext.packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY)
-    }
+    /**
+    The ActivityResultContract which should be used when launching this activity.
+    The launch argument is an Array<String> containing the categories to select from.
+    The return value is a Pair containing the selected category as a first element and the Uri pointing to the image as a second element
+     */
+    object AddPictureContract : ActivityResultContract<Array<String>, Pair<String, Uri>?>() {
+        override fun createIntent(context: Context, input: Array<String>?): Intent =
+            Intent(context, AddPictureActivity::class.java).putExtra("categories", input)
 
-    private fun setupRequestPermissionLauncher(): ActivityResultLauncher<String> {
-        return registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
-            if (isGranted) {
-                setupActivityLayout()
-                Toast.makeText(
-                    applicationContext,
-                    getString(R.string.AddPicture_permissionGrantedToast),
-                    Toast.LENGTH_SHORT
-                ).show()
+        override fun parseResult(resultCode: Int, result: Intent?): Pair<String, Uri>? {
+            return if (resultCode != Activity.RESULT_OK) {
+                null
             } else {
-                permissionNeededDialog()
+                val bundle = result!!.extras!!.get("result") as Bundle
+                Pair(bundle["category"] as String, bundle["image"] as Uri)
             }
         }
     }
 
-    private fun onSelectCategoryButton(v: View) {
+    @Suppress("UNUSED_PARAMETER")
+    private fun onSave(view: View) {
+        val returnIntent = Intent()
+        val bundle = Bundle().apply {
+            putString("category", chosenCategory)
+            putParcelable("image", capturedImageUri)
+        }
+        returnIntent.putExtra("result", bundle)
+        setResult(Activity.RESULT_OK, returnIntent)
+        finish()
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    private fun onTakePicture(view: View) {
+        val executor = Executors.newSingleThreadExecutor()
```

Comment:
What's the reason we need a brand new thread for this? Seems like all it does is `Log.e`